sql/parser/walk.go

```diff
@@ -23,14 +23,18 @@ import "fmt"
 // WalkExpr. The returned Expr replaces the pointer to the visited expression
 // in the parent node and can be used for rewriting expressions.
 type Visitor interface {
-	Visit(expr Expr) Expr
+	Visit(expr Expr, pre bool) (Visitor, Expr)
 }
 
 // WalkExpr traverses the nodes in an expression. It starts by calling
-// v.Visit(expr). It then recursively traverses the children nodes of the
-// expression returned by v.Visit().
+// v.Visit(expr, true). If the visitor returned by v.Pre(expr) is not nil it
```

Comment:
Well, it was, and then I added the `pre` argument to `Visit`. Fixed the comment.